runtime/src/mm/cpp/ObjectFactory.hpp

```diff
@@ -59,19 +60,15 @@ class ObjectFactoryStorage : private Pinned {
 
         Node() noexcept = default;
 
-        static KStdUniquePtr<Node> Create(size_t dataSize) noexcept {
+        template <typename Allocator>
+        static KStdUniquePtr<Node> Create(Allocator& allocator, size_t dataSize) noexcept {
             size_t dataSizeAligned = AlignUp(dataSize, DataAlignment);
             size_t totalAlignment = std::max(alignof(Node), DataAlignment);
             size_t totalSize = AlignUp(sizeof(Node) + dataSizeAligned, totalAlignment);
             RuntimeAssert(
                     DataOffset() + dataSize <= totalSize, "totalSize %zu is not enough to fit data %zu at offset %zu", totalSize, dataSize,
                     DataOffset());
-            void* ptr = konanAllocAlignedMemory(totalSize, totalAlignment);
-            if (!ptr) {
-                // TODO: Try doing GC first.
-                konan::consoleErrorf("Out of memory trying to allocate %zu. Aborting.\n", totalSize);
-                konan::abort();
-            }
+            void* ptr = allocator.Alloc(totalSize, totalAlignment);
```

Comment:
So the memory is allocated using `allocator.Alloc`, but to be deallocated with `KStdUniquePtr` delegating to `konanDestructInstance`? Looks like different levels of abstraction mixed.